### ISSUE TITLE
Forgroud Service crashing on android 14 ( API 34 )

### ISSUE THREAD

**Issue by Rohan487** [781flyingdutchman/background_downloader#249]
**Describe the bug**
App is crashing when using background downloader plugin in app which targets SDK 34 

**To Reproduce**
set target sdk to 34 on your app level build.gradle

**Expected behavior**
A workaround should be given on docs of background downloader plugin , but no workaround

**Logs**
![Screenshot 2024-02-23 210432](https://github.com/781flyingdutchman/background_downloader/assets/72985121/02439e20-af5e-4ffc-bbe1-6660e97ad16c)

**Additional context**
These infos may help 
https://developer.android.com/guide/components/foreground-services#fgs-prerequisites
https://stackoverflow.com/questions/76943771/foreground-service-crashing-on-android-14

**Comment by 781flyingdutchman** [comment]
Interesting - I was not aware of that. The stack overflow answer you reference has a comment that talks about this also being required for workers. Apparently you need to declare android:name="androidx.work.impl.foreground.SystemForegroundService" as follows:
```
<service
   android:name="androidx.work.impl.foreground.SystemForegroundService"
   android:foregroundServiceType="datasync"
   tools:node="merge" />
```

I am not 100% sure that the datasync service type is correct, but per this it looks like it: https://developer.android.com/develop/background-work/services/fg-service-types#data-sync (you may also have to add app permissions for datasync).

Can you try that and see if it works? I'm not in a place where I can test code for several days. If it works, I'll add it to the plugin manifest so you don't have to do it in the app, but I'm hoping this solves your problem for now.

**Comment by 781flyingdutchman** [comment]
Per [this](https://stackoverflow.com/questions/76949759/foregroundservicetype-0x00000001-is-not-a-subset-of-foregroundservicetype-attrib#:~:text=Thanks%20to%20%40sebastian%20I%20got%20the%20right,used%20in%20application%2C%20but%20WorkManager%20is%20used.) you also need to declare `<uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC"/>`, and it looks like I may have to change the code to use this in the foreground service call for API 34 and above. The article also says this issue wasn't there in beta versions of Android 34, which explains why I have not encountered this before.

**Comment by Rohan487** [comment]
Hey @781flyingdutchman,

I'm confident that the issue is solely caused by the Background Downloader plugin. As of now, I've only conducted tests in the foreground mode since my app typically deals with large files. I haven't implemented any additional plugins or native code apart from the Background Downloader, which interacts with workers or any foreground services.

I'll run the tests you suggested and keep you updated here.

**Comment by 781flyingdutchman** [comment]
Hi, I've added the necessary code change on the `foreground` branch of the repo. Can you please test it? Change your pubspec.yaml entry to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: foreground
```

and please note the new instructions: If targeting API 34 or greater, you must add to your `AndroidManifest.xml` a permission declaration `<uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC" />` and the foreground service type definition (under the `application` element):
  ```
  <service
    android:name="androidx.work.impl.foreground.SystemForegroundService"
    android:foregroundServiceType="dataSync"
    tools:node="merge" />
  ```

Can you please test this and let me know if this solves the problem on API 34? If so, I will publish this as a new version.

**Comment by 781flyingdutchman** [comment]
And note that the service definition includes `android:foregroundServiceType="dataSync"`, not `android:foregroundServiceType="datasync"` per my earlier post. Case matters, apparently

**Comment by 781flyingdutchman** [comment]
@Rohan487 can you take a look and let me know if these changes solve the problem? Thanks!

**Comment by 781flyingdutchman** [comment]
Implemented in V8.3.0
